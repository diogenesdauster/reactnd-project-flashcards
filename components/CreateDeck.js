import React, { Component } from "react";
import PropTypes from "prop-types";
import {
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  Platform,
  Alert,
  KeyboardAvoidingView
} from "react-native";
import { gray, lightblue3, white } from "../utils/colors";

class CreateDeck extends Component {
  static propTypes = {
    navigation: PropTypes.shape({ navigate: PropTypes.func.isRequired })
      .isRequired,
    createNewDeck: PropTypes.func.isRequired
  };

  state = {
    title: ""
  };
  _handleSubmitNewDeck = () => {
    const { createNewDeck, navigation } = this.props;
    const { title } = this.state;

    if (title) {
      createNewDeck(title);
      this.setState({ title: "" });
      navigation.navigate("DeckDetail", { title }, null, { key: "Home" });
    } else {
      Alert.alert("No Title", "Please write a title for adding a new Deck!!");
    }
  };
  render() {
    const { title } = this.state;
    return (
      <KeyboardAvoidingView style={styles.container}>
        <TextInput
          placeholder="Enter the title for your deck"
          style={Platform.OS === "ios" ? styles.inputIOS : styles.inputAndroid}
          onChangeText={title => this.setState({ title })}
          value={title}
        />
        <Text style={styles.label}>Title</Text>
        <TouchableOpacity
          style={styles.button}
          onPress={this._handleSubmitNewDeck}
        >
          <Text style={styles.buttonText}>Save</Text>
        </TouchableOpacity>
      </KeyboardAvoidingView>
    );
  }
}

export default CreateDeck;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 20
  },
  label: {
    paddingTop: 5,
    color: gray,
    fontSize: 20
  },
  inputIOS: {
    borderBottomWidth: 1,
    height: 37,
    padding: 10
  },
  inputAndroid: {
    height: 37,
    padding: 10
  },
  button: {
    width: 100,
    alignSelf: "flex-end"
  },
  buttonText: {
    backgroundColor: lightblue3,
    color: white,
    padding: 10,
    textAlign: "center"
  }
});
