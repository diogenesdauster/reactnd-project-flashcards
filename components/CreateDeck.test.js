import "react-native";
import React from "react";
import CreateDeck from "./CreateDeck";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

// static propTypes = {
//   navigation: PropTypes.shape({ navigate: PropTypes.func.isRequired })
//     .isRequired,
//   createNewDeck: PropTypes.func.isRequired
// };

it("renders CreateDeck correctly", () => {
  const mockNavigation = {
    navigate: jest.fn()
  };
  const mockCreateNewDeck = jest.fn();

  const tree = renderer
    .create(
      <CreateDeck
        navigation={mockNavigation}
        createNewDeck={mockCreateNewDeck}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
