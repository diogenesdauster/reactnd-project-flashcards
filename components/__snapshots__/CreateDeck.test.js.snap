// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CreateDeck correctly 1`] = `
<View
  enabled={true}
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "margin": 20,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter the title for your deck"
    style={
      Object {
        "borderBottomWidth": 1,
        "height": 37,
        "padding": 10,
      }
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#757575",
        "fontSize": 20,
        "paddingTop": 5,
      }
    }
  >
    Title
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "opacity": 1,
        "width": 100,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "#00BCD4",
          "color": "#FFFFFF",
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Save
    </Text>
  </View>
</View>
`;
